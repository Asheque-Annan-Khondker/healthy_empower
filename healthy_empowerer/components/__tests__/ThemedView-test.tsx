import * as React from 'react';
import renderer from 'react-test-renderer';

import { ThemedView } from '../ThemedView';

// Mock the useThemeColor hook
jest.mock('@/hooks/useThemeColor', () => ({
  useThemeColor: jest.fn(() => '#ffffff'), // Mocked color as white
}));

describe('ThemedView Component', () => {
  it('renders correctly with light and dark colors', () => {
    const tree = renderer
      .create(
        <ThemedView lightColor="#f0f0f0" darkColor="#333333">
          <></>
        </ThemedView>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('applies custom styles correctly', () => {
    const tree = renderer
      .create(
        <ThemedView lightColor="#f0f0f0" darkColor="#333333" style={{ padding: 10 }}>
          <></>
        </ThemedView>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});