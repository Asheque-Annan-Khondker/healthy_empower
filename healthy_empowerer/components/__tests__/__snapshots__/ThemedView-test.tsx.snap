// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedView Component applies custom styles correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
      },
      {
        "padding": 10,
      },
    ]
  }
/>
`;

exports[`ThemedView Component renders correctly with light and dark colors 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
      },
      undefined,
    ]
  }
/>
`;
